lib/msal-browser/src/cache/BrowserCacheManager.ts

```diff
@@ -1203,7 +1203,15 @@ export class BrowserCacheManager extends CacheManager {
     } | null {
         const key = `${Constants.CACHE_PREFIX}.${TemporaryCacheKeys.INTERACTION_STATUS_KEY}`;
         const value = this.getTemporaryCache(key, false);
-        return value ? JSON.parse(value) : null;
+        try {
+            return value ? JSON.parse(value) : null;
+        } catch (e) {
+            // Remove interaction and other temp keys when redirect begins on version <4.10.0 and completes on 4.10.0+
```

Comment:
This comment isn't really accurate here.  You're not checking versions explicitly, you're really saying if I can't parse the old state then let's clean it up for a subsequent go.  We can note for historic reasons that there we a break in 4.10.0 that needs this.

Reply:
Updated the comment